streams/src/main/java/org/apache/kafka/streams/processor/internals/ActiveTaskCreator.java

```diff
@@ -64,6 +65,9 @@
     private final Map<TaskId, StreamsProducer> taskProducers;
     private final StreamThread.ProcessingMode processingMode;
 
+    // tasks may be assigned for a NamedTopology that is not yet known by this host, and saved for later creation
+    private final Map<TaskId, Set<TopicPartition>>  unknownTasksToBeCreated = new HashMap<>();
```

Comment:
I think we discussed this already, but in case anyone else is wondering: the leader will always assign tasks based on its view of the current named topologies and topics, ie it does not check on individual subscriptions since the group is assumed to be eventually consistent in this regard. (Note this is actually no different than today; even if each instance of an app has a different input topic in their topology, they will all wind up receiving tasks for whichever topic the leader happened to have.)

Reply:
Yeah I left this comment before syncing with @ableegoldman offline, so it's better to clarify for others reading this PR :)